crates/vm/levm/src/precompiles.rs

```diff
@@ -1148,11 +1149,117 @@ fn point_evaluation(
 }
 
 pub fn bls12_g1add(
-    _calldata: &Bytes,
-    _gas_for_call: u64,
-    _consumed_gas: &mut u64,
+    calldata: &Bytes,
+    gas_for_call: u64,
+    consumed_gas: &mut u64,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    dbg!("entro", calldata.len(), consumed_gas.clone(), gas_for_call);
+    // Two inputs of 128 bytes are requiered
+    if calldata.len() != 256 {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    dbg!("pasa test length");
+
+    let mut first_point_x = calldata
+        .get(0..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    let mut first_point_y = calldata
+        .get(64..128)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    let mut second_point_x = calldata
+        .get(128..192)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    let mut second_point_y = calldata
+        .get(192..)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+
+    let zeros: [u8; 16] = [0_u8; 16];
+
+    // the first 16 bytes of any of the points MUST be all zeros
+    if first_point_x
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+    if first_point_y
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+    if second_point_x
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+    if second_point_y
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    first_point_x = first_point_x
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    first_point_y = first_point_y
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    second_point_x = second_point_x
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    second_point_y = second_point_y
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+
+    dbg!(first_point_x, first_point_y, second_point_x, second_point_y);
+
+    let mut first_g1_points = Vec::new();
```

Comment:
This is just one point
```suggestion
    let mut first_g1_point = Vec::new();
```